CI/COMPAT: TestExplore.test_vmin_vmax failing with latest folium

Failure with latest folium (not yet on main, only on PR with more recent build (https://github.com/geopandas/geopandas/actions/runs/3822563025/jobs/6502838246), so I suspect it the latest commit in folium, potentially https://github.com/python-visualization/folium/pull/1677)

```
__________________________ TestExplore.test_vmin_vmax __________________________
[gw1] linux -- Python 3.10.8 /usr/share/miniconda3/envs/test/bin/python3.10

self = <geopandas.tests.test_explore.TestExplore object at 0x7fcb19d188b0>

    def test_vmin_vmax(self):
        df = self.world.copy()
        df["range"] = range(len(df))
        m = df.explore("range", vmin=-100, vmax=1000)
        out_str = self._fetch_map_string(m)
        assert 'case"176":return{"color":"#3b528b","fillColor":"#3b528b"' in out_str
        assert 'case"119":return{"color":"#414287","fillColor":"#414287"' in out_str
        assert 'case"3":return{"color":"#482173","fillColor":"#482173"' in out_str
    
        # test 0
        df2 = self.nybb.copy()
        df2["values"] = df2["BoroCode"] * 10.0
        m = df2[df2["values"] >= 30].explore("values", vmin=0)
        out_str = self._fetch_map_string(m)
        assert 'case"1":return{"color":"#7ad151","fillColor":"#7ad151"' in out_str
>       assert 'case"2":return{"color":"#22a884","fillColor":"#22a884"' in out_str
E       assert 'case"2":return{"color":"#22a884","fillColor":"#22a884"' in '<!DOCTYPEhtml><html><head><metahttp-equiv="content-type"content="text/html;charset=UTF-8"/><script>L_NO_TOUCH=false;L_DISABLE_3D=false;</script><style>html,body{width:100%;height:100%;margin:0;padding:0;}</style><style>#map{position:absolute;top:0;bottom:0;right:0;left:0;}</style><scriptsrc="https://cdn.jsdelivr.net/npm/leaflet@1.9.3/dist/leaflet.js"></script><scriptsrc="https://code.jquery.com/jquery-1.12.4.min.js"></script><scriptsrc="https://cdn.jsdelivr.net/npm/bootstrap@5.2.2/dist/js/bootstrap.bundle.min.js"></script><scriptsrc="https://cdnjs.cloudflare.com/ajax/libs/Leaflet.awesome-markers/2.0.2/leaflet.awesome-markers.js"></script><linkrel="stylesheet"href="https://cdn.jsdelivr.net/npm/leaflet@1.9.3/dist/leaflet.css"/><linkrel="stylesheet"href="https://cdn.jsdelivr.net/npm/bootstrap@5.2.2/dist/css/bootstrap.min.css"/><linkrel="stylesheet"href="https://netdna.bootstrapcdn.com/bootstrap/3.0.0/css/bootstrap.min.css"/><linkrel="stylesheet"href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@6.2.0/css/all.min.css"/><linkrel="stylesheet"href="https://cdnjs.cloudflare.com/ajax/libs/Leaflet.awesome-markers/2.0.2/leaflet.awesome-markers.css"/><linkrel="stylesheet"href=...b1b9d9f06f3fd18f4.svg=d3.select(".legend.leaflet-control").append("svg").attr("id",\'legend\').attr("width",450).attr("height",40);color_map_670f019f9ce1ce9b1b9d9f06f3fd18f4.g=color_map_670f019f9ce1ce9b1b9d9f06f3fd18f4.svg.append("g").attr("class","key").attr("transform","translate(25,16)");color_map_670f019f9ce1ce9b1b9d9f06f3fd18f4.g.selectAll("rect").data(color_map_670f019f9ce1ce9b1b9d9f06f3fd18f4.color.range().map(function(d,i){return{x0:i?color_map_670f019f9ce1ce9b1b9d9f06f3fd18f4.x(color_map_670f019f9ce1ce9b1b9d9f06f3fd18f4.color.domain()[i-1]):color_map_670f019f9ce1ce9b1b9d9f06f3fd18f4.x.range()[0],x1:i<color_map_670f019f9ce1ce9b1b9d9f06f3fd18f4.color.domain().length?color_map_670f019f9ce1ce9b1b9d9f06f3fd18f4.x(color_map_670f019f9ce1ce9b1b9d9f06f3fd18f4.color.domain()[i]):color_map_670f019f9ce1ce9b1b9d9f06f3fd18f4.x.range()[1],z:d};})).enter().append("rect").attr("height",40-30).attr("x",function(d){returnd.x0;}).attr("width",function(d){returnd.x1-d.x0;}).style("fill",function(d){returnd.z;});color_map_670f019f9ce1ce9b1b9d9f06f3fd18f4.g.call(color_map_670f019f9ce1ce9b1b9d9f06f3fd18f4.xAxis).append("text").attr("class","caption").attr("y",21).text("values");</script></html>'
```

--- a/geopandas/tests/test_explore.py
+++ b/geopandas/tests/test_explore.py
@@ -14,6 +14,7 @@ import matplotlib.colors as colors  # noqa
 from branca.colormap import StepColormap  # noqa
 
 BRANCA_05 = Version(branca.__version__) > Version("0.4.2")
+FOLIUM_G_014 = Version(folium.__version__) > Version("0.14.0")
 
 
 class TestExplore:
@@ -524,8 +525,14 @@ class TestExplore:
         df2["values"] = df2["BoroCode"] * 10.0
         m = df2[df2["values"] >= 30].explore("values", vmin=0)
         out_str = self._fetch_map_string(m)
-        assert 'case"1":return{"color":"#7ad151","fillColor":"#7ad151"' in out_str
-        assert 'case"2":return{"color":"#22a884","fillColor":"#22a884"' in out_str
+        if FOLIUM_G_014:
+            assert 'case"0":return{"color":"#fde725","fillColor":"#fde725"' in out_str
+            assert 'case"1":return{"color":"#7ad151","fillColor":"#7ad151"' in out_str
+            assert 'default:return{"color":"#22a884","fillColor":"#22a884"' in out_str
+        else:
+            assert 'case"1":return{"color":"#7ad151","fillColor":"#7ad151"' in out_str
+            assert 'case"2":return{"color":"#22a884","fillColor":"#22a884"' in out_str
+            assert 'default:return{"color":"#fde725","fillColor":"#fde725"' in out_str
 
         df2["values_negative"] = df2["BoroCode"] * -10.0
         m = df2[df2["values_negative"] <= 30].explore("values_negative", vmax=0)

